components/script/dom/promisenativehandler.rs

```diff
@@ -14,6 +14,10 @@ use crate::dom::globalscope::GlobalScope;
 use crate::realms::InRealm;
 use crate::script_runtime::{CanGc, JSContext as SafeJSContext};
 
+/// Types that implement the `Callback` trait follow the same rooting requirements
+/// as types that use the `#[dom_struct]` attribute.
+/// Prefer storing `Dom<T>` members inside them instead of `DomRoot<T>`
+/// to minimize redundant work by the garbage collector.
 pub trait Callback: JSTraceable + MallocSizeOf {
     fn callback(&self, cx: SafeJSContext, v: HandleValue, realm: InRealm, can_gc: CanGc);
 }
```

Comment:
> follow the same rooting requirements as types that use the `#[dom_struct]`

So we should treat it as it's marked with `#[crown::unrooted_must_root_lint::must_root]`? Maybe we should impl this in crown, it could work similar to https://github.com/servo/servo/pull/34163.

Reply:
Not necessarily. A callback doesn't have to contain values that can be GCed, but it is stored in a way that is reachable by the GC, so it is safe to store such values inside it without additional rooting.

Reply:
Ah, so it is JSManaged that must be traced (and every struct that uses it must also be traced). Relevant issue for safety of this: https://github.com/servo/servo/issues/32102.